### PR TITLE
[ENG-3054] App update, fixes and improvements

### PR DESCRIPTION
**Pull request by kennyshittu**
This PR adds the following changes :

- Fixes DB connection issue and other minor bugs
- Removes unused problematic DB migration files 
- Adds Slack manifest file for easier setup
- Update contribution doc
- Test update and improvements

Ticket : https://dataworld.atlassian.net/browse/ENG-3054

### CODE REVIEW

File: server/api/slack.js [hunk 1]
```diff
@@ -177,7 +180,7 @@ const dismissAuthRequiredMessage = async (responseUrl) => {
   }
 };
 
-const startUnfurlAssociation = async (nonce, botAccessToken, channel, slackUserId, messageTs, teamAccessToken, teamId) => {
```
Comment (kennyshittu) [inline server/api/slack.js:180]: removed unused teamId

File: server/controllers/unfurl.js [hunk 2]
```diff
@@ -535,9 +535,8 @@ const handleLinkSharedEvent = async (event, teamId, serverBaseUrl) => {
       if (isAssociated) {
         // User is associated, carry on and unfold url
         let token = user.dwAccessToken;
-        const teamToken = process.env.SLACK_TEAM_TOKEN || team.accessToken;
-
-        const botToken = await getBotAccessTokenForTeam(team.teamId);
+        // const teamToken = process.env.SLACK_TEAM_TOKEN || team.accessToken;
```
Comment (kennyshittu) [inline server/controllers/unfurl.js:538]: teamToken is unused, remove it completly

File: server/models/index.js [hunk 3]
```diff
@@ -28,16 +28,21 @@ const db = {};
 
 let sequelize;
 if (process.env.DATABASE_URL) {
-  sequelize = new Sequelize(process.env.DATABASE_URL, { logging : false });
+  sequelize = new Sequelize(process.env.DATABASE_URL, { logging : false,  dialectOptions: {
+    ssl: {
+      require: true, // This will enable SSL
```
Comment (kennyshittu) [inline server/models/index.js:33]: Heroku postgres addon requires ssl

File: server/controllers/unfurl.js [hunk 4]
```diff
@@ -562,7 +559,7 @@ const handleLinkSharedEvent = async (event, teamId, serverBaseUrl) => {
               event.message_ts,
               event.channel,
               unfurls,
-              botToken
+              teamToken
```
Comment (saltshop-boy) [inline server/controllers/unfurl.js:562]: Is this change related to a specific bug fix?

**Review by saltshop-boy — COMMENTED** [review]
@kennyshittu, Looks good! I have a couple of questions.

File: server/controllers/unfurl.js (hunk 4 above)
Comment (kennyshittu) [reply server/controllers/unfurl.js:562]: Yes, while testing with the staging app we have on heroku link unfurl was failing with the error below when we use bot access token

```
2024-04-11T21:45:42.663588+00:00 app[web.1]: [0mPOST /api/v1/unfurl/action [32m200 [0m0.906 ms - 30[0m
2024-04-11T21:45:42.906196+00:00 app[web.1]: (node:54) UnhandledPromiseRejectionWarning: Error: An API error occurred: not_allowed_token_type
2024-04-11T21:45:42.906199+00:00 app[web.1]:     at platformErrorFromResult (/app/node_modules/@slack/web-api/dist/errors.js:51:33)
2024-04-11T21:45:42.906200+00:00 app[web.1]:     at WebClient.apiCall (/app/node_modules/@slack/web-api/dist/WebClient.js:171:56)
2024-04-11T21:45:42.906201+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:95:5)
2024-04-11T21:45:42.906201+00:00 app[web.1]:     at emitUnhandledRejectionWarning (internal/process/promises.js:168:15)
2024-04-11T21:45:42.906201+00:00 app[web.1]:     at processPromiseRejections (internal/process/promises.js:247:11)
2024-04-11T21:45:42.906202+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:96:32)
2024-04-11T21:45:42.906207+00:00 app[web.1]: (node:54) Error: An API error occurred: not_allowed_token_type
2024-04-11T21:45:42.906207+00:00 app[web.1]:     at platformErrorFromResult (/app/node_modules/@slack/web-api/dist/errors.js:51:33)
2024-04-11T21:45:42.906207+00:00 app[web.1]:     at WebClient.apiCall (/app/node_modules/@slack/web-api/dist/WebClient.js:171:56)
2024-04-11T21:45:42.906208+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:95:5)
```
That said, I also noticed that using bot access token works when I test locally, so it could be that old both access tokens (like the one we have on staging which was generated in 2018) are not valid for the `chat.unfurl()` method. This would explain why this was originally implemented to us team access token before it was changed in this [PR](https://github.com/datadotworld/slack-app/pull/78/files#diff-e64e471132737dafd3fe025122fa1ea5ecda3e449128c413e1ef024ef99089b4R540).

Hence, to ensure the link unfurling don't break in prod it's better to use team access token which is guaranteed to work for both old and newly generated tokens.

File: server/config/config.json [hunk 5]
```diff
@@ -1,13 +1,16 @@
 {
   "development": {
-    "username": "PG_USERNAME",
-    "password": "PG_PASSWORD",
-    "database": "PG_DATABASE",
```
[the remark this answers is not in the document]
Comment (kennyshittu) [reply server/config/config.json:5]: Yes, the database URL would have all the info required to connect. Setting the following env variables instead of a DB URL would also work 

DB_USER=***
DB_NAME=***
DB_PASS=***

**Review by saltshop-boy — APPROVED** [review]

File: server/controllers/unfurl.js (hunk 4 above)
Comment (kennyshittu) [reply server/controllers/unfurl.js:562]: @saltshop-boy On second thought, I think we should use bot tokens since thats the slack preferred way of sending  messages in the new API. Once we republish the app, existing customers can reinstall the app. Another down side to using team access token is that, if the user that installed the app is removed from the workspace, the team access token would become invalid, so it's better to not depend on it.

**Review by saltshop-boy — APPROVED** [review]
